### DISCUSSION TITLE
Does pymssql automatically perform input sanitization for parameterized queries?

### DISCUSSION THREAD

**Discussion by annahansen229** [pymssql/pymssql#863] [Q&A]
Been looking through the docs and cannot find this mentioned explicitly. 

For either `cursor.execute(sql_string, (param1, param2))` or `cursor.callproc(proc_name, (param1, param2))`, does pymssql sanitize `param1` and `param2` or do I need to sanitize them first?

**Comment by termim** [comment]
What you mean by sanitize?

**Reply by annahansen229** [reply]
Removing special characters from the string.

**Reply by termim** [reply]
If you do not need them inserted, then you should remove them yourself.

**Comment by gordthompson** [comment]
Yes, pymssql does sanitize parameter values. For example, if we do

```python
sql = "SELECT id FROM client WHERE last_name = %s"
cursor.execute(sql, ("O'Connell"))
```

the statement received by the server is

```sql
SELECT id FROM client WHERE last_name = N'O''Connell'
```

**Reply by termim** [reply]
In this sense yes, pymssql does quote some characters. But it never removes any characters from the user data. If it does, it is a bug.
